espnet2/spk/loss/aamsoftmax.py

```diff
@@ -0,0 +1,72 @@
+#! /usr/bin/python
+# -*- encoding: utf-8 -*-
+# code from https://github.com/clovaai/voxceleb_trainer/blob/master/loss/aamsoftmax.py
+# Adapted from https://github.com/wujiyang/Face_Pytorch (Apache License)
+
+import math
+
+import numpy
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+
+class AAMSoftmax(nn.Module):
+    """
+    Additive angular margin softmax.
+
+    Paper: Deng, Jiankang, et al. "Arcface: Additive angular margin loss for
+    deep face recognition." Proceedings of the IEEE/CVF conference on computer
+    vision and pattern recognition. 2019.
+    """
+
+    def __init__(
+        self, nOut, nClasses, margin=0.3, scale=15, easy_margin=False, **kwargs
+    ):
+        super().__init__()
+
+        self.test_normalize = True
+
+        self.m = margin
+        self.s = scale
+        self.in_feats = nOut
+        self.weight = torch.nn.Parameter(
+            torch.FloatTensor(nClasses, nOut), requires_grad=True
+        )
+        self.ce = nn.CrossEntropyLoss()
+        nn.init.xavier_normal_(self.weight, gain=1)
```

Comment:
I am not sure if the initialized parameters here will be re-initialized in https://github.com/espnet/espnet/blob/0bc12312609b51799a8855ea1054c3873f77c640/espnet2/tasks/spk.py#L301

Reply:
I checked the code and I think is args.init is given, even this part will follow that.